covid19dp_submission/nextflow/submission_workflow_template.nf

```diff
@@ -0,0 +1,81 @@
+process download_snapshot {{
+input:
+val flag from true
+output:
+val true into download_snapshot_success
+shell:
+'''
+(export PYTHONPATH="{app[python][script_path]}" && {app[python][interpreter]} -m steps.download_snapshot --download-url {submission[download_url]} --snapshot-name {submission[snapshot_name]} --download-target-dir {submission[download_target_dir]}) >> {submission[log_dir]}/download_snapshot.log 2>&1
```

Comment:
Shell blocks are executed verbatim and could be executed on different machines. They are not inherited. Environment variable can be set via [input env](https://www.nextflow.io/docs/latest/process.html#input-of-type-env) but the repetition will still unfortunately happen across all the process definitions and hence won't be that different from what we currently have.

Reply:
Since you have `setup.py`, you should be able to `pip install` the module into the exact python binary you're passing to Nextflow, then you shouldn't need to `export PYTHONPATH` for each process.

Reply:
I agree that the script could be deployed as executables which would alleviate the need for these `PYTHONPATH`.
Couple things needed for that: 
 - set the shebang at the top of each script. 
 - add each step to list of scripts in `setup.py`
 - If they end up in the global path for `eva_etl`, we should think about the name of the scripts so that they don't conflict with others.

Reply:
Added as part of https://www.ebi.ac.uk/panda/jira/browse/EVA-2536